### ISSUE TITLE
simple upgrade command

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#2598]
## Tell us about the problem you're trying to solve
I would love to be able to upgrade Airbyte by running something like `./upgrade_airbyte.sh` so I don't have to follow a non-trivial path like described in https://docs.airbyte.io/tutorials/upgrading-airbyte. 

## Describe the solution you’d like
`./upgrade_airbyte.sh`. I will also accept `./upgrade_airbyte`.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368155896011) by [Unito](https://www.unito.io)

**Comment by jrhizor** [comment]
I want this to detect that a version change occurred and run automatically, so all users have to do is `docker-compose down`, `git pull`, `docker-compose up`.

**Comment by jrhizor** [comment]
To add to my comment above, automatic upgrades could happen within the server at startup:
* The server would be responsible for migrations.
* It would also be responsible for db initialization.
* Before updating, it would save a backup snapshot of the configuration so it'd be possible to revert.
* We would need to disable the API and UI or make it read only during the upgrade process.

**Comment by akoshterek** [comment]
Also an idea: 

- Export configuration via UI/API
- install fresh Airbyte version (or spin up a new VM)
- Import the exported file back via UI/API
- Migration happens automatically

**Comment by cgardens** [comment]
Adding this as a dependency of Airbyte Cloud. I think the consensus is this process has to be tigher.

**Comment by cgardens** [comment]
I'm going to summarize the work / goals for this issue here. It will mostly be copying and pasting content from previous posts. Just want it to be super clear and all in one spot.

# Requirements
* As a user, when I want to upgrade Airbyte, all I should have to do is `docker-compose down`, `git pull`, `docker-compose up`. Note: `git pull` essentially just means bumping the version in the .env file to the desired version.
* The upgrade process should be "atomic" (or as close it at possible) and "all or nothing". If a migration fails, then the user should always be able to retreat back to the last version that was working for them safely. For example, if the user was on version 2 and was trying to upgrade to version 4. If the migration from version 2 to 3 succeeds and 3 to 4 fails, then the user should be allowed to go back to version 2. 

## Non-Goals
* Downgrading. We assuming that this upgrade process only goes up. If a user was on version 2 and sets versions to 1, we should fail to start.

## Suggestions
* Conceptually, none of the other components in Airbyte are really supposed to start until the server is up and running. De facto this is true in the current setup because a while loop in the main method of the scheduler. The current implementation is brittle and will probably needs to be adjusted. One approach that @jrhizor and I have discussed is that all components wait for the server to start up (perhaps waiting on a health check instead of the current mechanism). This way the migrations can run (if needed) as part of the start up of the server.
